src/commons/controlBar/ControlBarToggleFolderModeButton.tsx

```diff
@@ -8,19 +8,23 @@ type Props = {
   isFolderModeEnabled: boolean;
   isSessionActive: boolean;
   isPersistenceActive: boolean;
+  isSupportedSource: boolean;
   toggleFolderMode: () => void;
 };
 
 export const ControlBarToggleFolderModeButton: React.FC<Props> = ({
   isFolderModeEnabled,
   isSessionActive,
   isPersistenceActive,
+  isSupportedSource,
   toggleFolderMode
 }) => {
   const tooltipContent = isSessionActive
     ? 'Currently unsupported while a collaborative session is active'
     : isPersistenceActive
     ? 'Currently unsupported while a persistence method is active'
+    : !isSupportedSource
+    ? 'Folder mode is not available for this version of Source'
     : `${isFolderModeEnabled ? 'Disable' : 'Enable'} Folder mode`;
   return (
     <Tooltip content={tooltipContent}>
```

Comment:
I'd argue that if the language does not support it, the button should just be hidden (not disabled), like on the Playground.

Reply:
What do you think?